cacerts/detect.go

```diff
@@ -0,0 +1,65 @@
+package cacerts
+
+import (
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/buildpacks/libcnb"
+	"github.com/paketo-buildpacks/libpak/bard"
+)
+
+type Detect struct {
+	Logger bard.Logger
+}
+
+const (
+	BindingType = "ca-certificates"
+)
+
+// Detect always passes and optionally provides ca-certificates. If there is a binding of
+// type "ca-certficates" Detect also requires ca-certificates and provides an array of cerficate paths in the
+// plan entry metadata.
+func (d Detect) Detect(context libcnb.DetectContext) (libcnb.DetectResult, error) {
+	paths := getsCertsFromBindings(context.Platform.Bindings)
+	if len(paths) > 0 {
+		return libcnb.DetectResult{Pass: true, Plans: []libcnb.BuildPlan{{
+			Provides: []libcnb.BuildPlanProvide{
+				{Name: "ca-certificates"},
+			},
+			Requires: []libcnb.BuildPlanRequire{
+				{
+					Name: "ca-certificates",
+					Metadata: map[string]interface{}{
+						"paths": paths,
+					},
+				},
+			},
+		}}}, nil
+	}
+
+	return libcnb.DetectResult{
+		Pass: true,
+		Plans: []libcnb.BuildPlan{
+			{
+				Provides: []libcnb.BuildPlanProvide{
+					{Name: "ca-certificates"},
+				},
+			},
+			{}, // always contribute runtime helper
+		},
+	}, nil
+}
+
+func getsCertsFromBindings(bindings libcnb.Bindings) []string {
+	var paths []string
+	for _, bind := range bindings {
+		if strings.TrimSpace(strings.ToLower(bind.Type)) == strings.TrimSpace(strings.ToLower(BindingType)) {
```

Comment:
> since it doesn't support the old style binding for example

I think `Type` must be populated with the value of `provider` for the old style bindings when the Binding struct is initialized, otherwise the resolved would not work properly https://github.com/paketo-buildpacks/libpak/blob/main/binding.go#L37-L41

I went to use the `BindingResolver` initially but it returns an error when there are multiple bindings of the given type, and in this case there is no particular reason to force users to have exactly one binding.

Reply:
Resolved with a update to `libpak` https://github.com/paketo-buildpacks/libpak/pull/22